analysis/descriptives/diabetes-follow-up-analysis.R

```diff
@@ -0,0 +1,269 @@
+# Additional analysis specific to post-covid diabetes project
+# Using data generated in the "generate_study_population_diabetes_analysis" study definition
+# Aim: Investigate how many of those diagnosed with type 2 diabetes following a covid-19 infection were still being treated or had elevated HbA1c levels
+## =============================================================================
+
+############################################
+# Load relevant libraries and read in data #
+############################################
+
+#Load libraries using pacman
+pacman::p_load(dplyr,tictoc,readr,stringr,tidyr,ggplot2,jsonlite,here,arrow)
+
+#clear memory
+rm(list=ls())
+
+# Specify command arguments ----------------------------------------------------
+
+args <- commandArgs(trailingOnly=TRUE)
+
+if(length(args)==0){
+  # use for interactive testing
+  cohort_name <- "prevax"
+} else {
+  cohort_name <- args[[1]]
+}
+
+fs::dir_create(here::here("output", "not-for-review"))
+fs::dir_create(here::here("output", "review", "descriptives"))
+
+# STUDY END DATES
+
+study_dates <- jsonlite::fromJSON("output/study_dates.json")
+
+if (cohort_name %in% c("vax","unvax"))
+{
+  #These are the study start and end dates for the Delta era
+  cohort_start_date <- as.Date(study_dates$delta_date)
+  cohort_end_date <- as.Date(study_dates$omicron_date)
+}else if (cohort_name == "prevax") {
+  cohort_start_date <- as.Date(study_dates$pandemic_start)
+  cohort_end_date <- as.Date(study_dates$all_eligible)
+}
+
+###############################################
+# DEFINE FUNCTION -----------------------------
+###############################################
+
+diabetes_post_hoc <- function(cohort_name){
+  
+# Load relevant data
+input <- arrow::read_feather(file = paste0("output/input_",cohort_name,"_diabetes_analysis.feather"))
+
+# summarise df
+summary(input)
+
+# Restrict data only to those that had a diagnosis of type 2 diabetes following a COVID-19 infection
+
+input_4 <- input %>%
+  dplyr::filter( !is.na(out_date_t2dm) &  !is.na(exp_date_covid19_confirmed)) %>%
+  # keep those where t2dm is after infection
+  rowwise() %>%
+  mutate(keep = ifelse(out_date_t2dm < exp_date_covid19_confirmed, FALSE, TRUE)) %>%
+  ungroup() %>%
+  dplyr::filter(keep == TRUE) %>%
+  dplyr::select(-c(keep))
+
+# summarise df
+summary(input_4)
+
+# calculate new end date
+
+input_4$cohort_end_date <- cohort_end_date
+input_4$end_date <- apply(input_4[,c("death_date", "dereg_date", "cohort_end_date")],1, min,na.rm=TRUE)
+input_4$end_date <- as.Date(input_4$end_date)
+
+# Get N with 4 months follow up (those with an end date >= 4 months from t2dm)
+
+input_4 <- input_4 %>% 
+  rowwise() %>%
+  mutate(start_end_diff = as.numeric(difftime(end_date, out_date_t2dm, units = "days"))) %>%
+  ungroup() %>%
+  mutate(start_end_diff_months = start_end_diff/30.417) %>%
+  mutate(follow_4mth = ifelse(start_end_diff_months >= 4, TRUE, FALSE))
```

Comment:
you could add follow_12mth column here instead of repeating the whole code for 12 mth

Reply:
Yes I understand what you mean, good point. But I only want to do the 12 month for prevax only `if (cohort_name == "prevax"){` which is why I just did it separately at the bottom. We may not even keep the 12 month analysis in (will see what the collaborators say), so will see what they say first and then go from there.